client/sql/sql_test.go

```
@@ -42,3 +48,299 @@
+			if tt.want.Protocol != "" {
+				assert.Equal(t, tt.want.Protocol, got.Protocol)
+			}
+		})
+	}
+}
+
+func TestOpen_InvalidDriver(t *testing.T) {
+	db, err := Open("nonexistent-driver", "invalid-dsn")
+
+	// Should return error for unknown driver
+	assert.Error(t, err)
```

golangci-lint (2.5.0)

[error] require-error: for error assertions use require

(testifylint)